### DISCUSSION TITLE
How should I implement user-defined properties for Body?

### DISCUSSION THREAD

**Discussion by saulo308** [jrouwe/JoltPhysics#858] [Q&A] [answered]
Hey guys, pretty new to physics engines here.

How should I implement user-defined properties for the Body class. In this case, I want every Body to have a custom enum and a property from this type to represent a Body status according to the game I'm implementing. How I should handle this? 

a) I was thinking of modifying the Body class and re-compile joltphysics, but for some reason I don't know if this is the best practise. 
b) I have a whole implementation that extends JoltPhysics using its headers that I build and ship. I was wondering if I could inherit from the Body class and have my own derived class with the modifications. However, I can't seem to find any examples of creating a custom Body using the BodyInterface and BodyCreationSettings. Thus, I also don't know if this is possible.

Thanks in advance!

**Comment by jrouwe** [comment, marked as answer]
Normally you would store a pointer to your own game object using [Body::SetUserData](https://jrouwe.github.io/JoltPhysics/class_body.html#a3fff2f601caed12dd89c6d7fcb04dad6) and store your properties there.

You cannot inherit from the Body class as it is placed in the same memory block as the MotionProperties object for cache locality. The Body class has been carefully layed out in memory, so adding your own stuff is not advised.

**Reply by saulo308** [reply]
Thank you! That will do the trick.
